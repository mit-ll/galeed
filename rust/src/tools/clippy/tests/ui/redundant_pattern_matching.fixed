// run-rustfix

#![warn(clippy::all)]
#![warn(clippy::redundant_pattern_matching)]
#![allow(
    clippy::unit_arg,
    unused_must_use,
    clippy::needless_bool,
    clippy::match_like_matches_macro,
    deprecated
)]

fn main() {
    let result: Result<usize, usize> = Err(5);
    if result.is_ok() {}

    if Ok::<i32, i32>(42).is_ok() {}

    if Err::<i32, i32>(42).is_err() {}

    if None::<()>.is_none() {}

    if Some(42).is_some() {}

    if Some(42).is_some() {
        foo();
    } else {
        bar();
    }

    while Some(42).is_some() {}

    while Some(42).is_none() {}

    while None::<()>.is_none() {}

    while Ok::<i32, i32>(10).is_ok() {}

    while Ok::<i32, i32>(10).is_err() {}

    let mut v = vec![1, 2, 3];
    while v.pop().is_some() {
        foo();
    }

    if Ok::<i32, i32>(42).is_ok() {}

    if Err::<i32, i32>(42).is_err() {}

    if None::<i32>.is_none() {}

    if Some(42).is_some() {}

    if let Ok(x) = Ok::<i32, i32>(42) {
        println!("{}", x);
    }

    Ok::<i32, i32>(42).is_ok();

    Ok::<i32, i32>(42).is_err();

    Err::<i32, i32>(42).is_err();

    Err::<i32, i32>(42).is_ok();

    Some(42).is_some();

    None::<()>.is_none();

    let _ = None::<()>.is_none();

    let _ = if Ok::<usize, ()>(4).is_ok() { true } else { false };

    let opt = Some(false);
    let x = if opt.is_some() { true } else { false };
    takes_bool(x);

    issue5504();
    issue5697();

    let _ = if gen_opt().is_some() {
        1
    } else if gen_opt().is_none() {
        2
    } else if gen_res().is_ok() {
        3
    } else if gen_res().is_err() {
        4
    } else {
        5
    };
}

fn gen_opt() -> Option<()> {
    None
}

fn gen_res() -> Result<(), ()> {
    Ok(())
}

fn takes_bool(_: bool) {}

fn foo() {}

fn bar() {}

macro_rules! m {
    () => {
        Some(42u32)
    };
}

fn issue5504() {
    fn result_opt() -> Result<Option<i32>, i32> {
        Err(42)
    }

    fn try_result_opt() -> Result<i32, i32> {
        while r#try!(result_opt()).is_some() {}
        if r#try!(result_opt()).is_some() {}
        Ok(42)
    }

    try_result_opt();

    if m!().is_some() {}
    while m!().is_some() {}
}

// None of these should be linted because none of the suggested methods
// are `const fn` without toggling a feature.
const fn issue5697() {
    if let Ok(_) = Ok::<i32, i32>(42) {}

    if let Err(_) = Err::<i32, i32>(42) {}

    if let Some(_) = Some(42) {}

    if let None = None::<()> {}

    while let Ok(_) = Ok::<i32, i32>(10) {}

    while let Err(_) = Ok::<i32, i32>(10) {}

    while let Some(_) = Some(42) {}

    while let None = None::<()> {}

    match Ok::<i32, i32>(42) {
        Ok(_) => true,
        Err(_) => false,
    };

    match Err::<i32, i32>(42) {
        Ok(_) => false,
        Err(_) => true,
    };
    match Some(42) {
        Some(_) => true,
        None => false,
    };

    match None::<()> {
        Some(_) => false,
        None => true,
    };
}
