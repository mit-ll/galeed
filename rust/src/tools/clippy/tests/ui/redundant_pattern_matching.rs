// run-rustfix

#![warn(clippy::all)]
#![warn(clippy::redundant_pattern_matching)]
#![allow(
    clippy::unit_arg,
    unused_must_use,
    clippy::needless_bool,
    clippy::match_like_matches_macro,
    deprecated
)]

fn main() {
    let result: Result<usize, usize> = Err(5);
    if let Ok(_) = &result {}

    if let Ok(_) = Ok::<i32, i32>(42) {}

    if let Err(_) = Err::<i32, i32>(42) {}

    if let None = None::<()> {}

    if let Some(_) = Some(42) {}

    if let Some(_) = Some(42) {
        foo();
    } else {
        bar();
    }

    while let Some(_) = Some(42) {}

    while let None = Some(42) {}

    while let None = None::<()> {}

    while let Ok(_) = Ok::<i32, i32>(10) {}

    while let Err(_) = Ok::<i32, i32>(10) {}

    let mut v = vec![1, 2, 3];
    while let Some(_) = v.pop() {
        foo();
    }

    if Ok::<i32, i32>(42).is_ok() {}

    if Err::<i32, i32>(42).is_err() {}

    if None::<i32>.is_none() {}

    if Some(42).is_some() {}

    if let Ok(x) = Ok::<i32, i32>(42) {
        println!("{}", x);
    }

    match Ok::<i32, i32>(42) {
        Ok(_) => true,
        Err(_) => false,
    };

    match Ok::<i32, i32>(42) {
        Ok(_) => false,
        Err(_) => true,
    };

    match Err::<i32, i32>(42) {
        Ok(_) => false,
        Err(_) => true,
    };

    match Err::<i32, i32>(42) {
        Ok(_) => true,
        Err(_) => false,
    };

    match Some(42) {
        Some(_) => true,
        None => false,
    };

    match None::<()> {
        Some(_) => false,
        None => true,
    };

    let _ = match None::<()> {
        Some(_) => false,
        None => true,
    };

    let _ = if let Ok(_) = Ok::<usize, ()>(4) { true } else { false };

    let opt = Some(false);
    let x = if let Some(_) = opt { true } else { false };
    takes_bool(x);

    issue5504();
    issue5697();

    let _ = if let Some(_) = gen_opt() {
        1
    } else if let None = gen_opt() {
        2
    } else if let Ok(_) = gen_res() {
        3
    } else if let Err(_) = gen_res() {
        4
    } else {
        5
    };
}

fn gen_opt() -> Option<()> {
    None
}

fn gen_res() -> Result<(), ()> {
    Ok(())
}

fn takes_bool(_: bool) {}

fn foo() {}

fn bar() {}

macro_rules! m {
    () => {
        Some(42u32)
    };
}

fn issue5504() {
    fn result_opt() -> Result<Option<i32>, i32> {
        Err(42)
    }

    fn try_result_opt() -> Result<i32, i32> {
        while let Some(_) = r#try!(result_opt()) {}
        if let Some(_) = r#try!(result_opt()) {}
        Ok(42)
    }

    try_result_opt();

    if let Some(_) = m!() {}
    while let Some(_) = m!() {}
}

// None of these should be linted because none of the suggested methods
// are `const fn` without toggling a feature.
const fn issue5697() {
    if let Ok(_) = Ok::<i32, i32>(42) {}

    if let Err(_) = Err::<i32, i32>(42) {}

    if let Some(_) = Some(42) {}

    if let None = None::<()> {}

    while let Ok(_) = Ok::<i32, i32>(10) {}

    while let Err(_) = Ok::<i32, i32>(10) {}

    while let Some(_) = Some(42) {}

    while let None = None::<()> {}

    match Ok::<i32, i32>(42) {
        Ok(_) => true,
        Err(_) => false,
    };

    match Err::<i32, i32>(42) {
        Ok(_) => false,
        Err(_) => true,
    };
    match Some(42) {
        Some(_) => true,
        None => false,
    };

    match None::<()> {
        Some(_) => false,
        None => true,
    };
}
