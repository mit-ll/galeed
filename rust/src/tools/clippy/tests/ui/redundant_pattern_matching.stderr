error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:15:12
   |
LL |     if let Ok(_) = &result {}
   |     -------^^^^^---------- help: try this: `if result.is_ok()`
   |
   = note: `-D clippy::redundant-pattern-matching` implied by `-D warnings`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:17:12
   |
LL |     if let Ok(_) = Ok::<i32, i32>(42) {}
   |     -------^^^^^--------------------- help: try this: `if Ok::<i32, i32>(42).is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching.rs:19:12
   |
LL |     if let Err(_) = Err::<i32, i32>(42) {}
   |     -------^^^^^^---------------------- help: try this: `if Err::<i32, i32>(42).is_err()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:21:12
   |
LL |     if let None = None::<()> {}
   |     -------^^^^------------- help: try this: `if None::<()>.is_none()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:23:12
   |
LL |     if let Some(_) = Some(42) {}
   |     -------^^^^^^^----------- help: try this: `if Some(42).is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:25:12
   |
LL |     if let Some(_) = Some(42) {
   |     -------^^^^^^^----------- help: try this: `if Some(42).is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:31:15
   |
LL |     while let Some(_) = Some(42) {}
   |     ----------^^^^^^^----------- help: try this: `while Some(42).is_some()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:33:15
   |
LL |     while let None = Some(42) {}
   |     ----------^^^^----------- help: try this: `while Some(42).is_none()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:35:15
   |
LL |     while let None = None::<()> {}
   |     ----------^^^^------------- help: try this: `while None::<()>.is_none()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:37:15
   |
LL |     while let Ok(_) = Ok::<i32, i32>(10) {}
   |     ----------^^^^^--------------------- help: try this: `while Ok::<i32, i32>(10).is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching.rs:39:15
   |
LL |     while let Err(_) = Ok::<i32, i32>(10) {}
   |     ----------^^^^^^--------------------- help: try this: `while Ok::<i32, i32>(10).is_err()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:42:15
   |
LL |     while let Some(_) = v.pop() {
   |     ----------^^^^^^^---------- help: try this: `while v.pop().is_some()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:58:5
   |
LL | /     match Ok::<i32, i32>(42) {
LL | |         Ok(_) => true,
LL | |         Err(_) => false,
LL | |     };
   | |_____^ help: try this: `Ok::<i32, i32>(42).is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching.rs:63:5
   |
LL | /     match Ok::<i32, i32>(42) {
LL | |         Ok(_) => false,
LL | |         Err(_) => true,
LL | |     };
   | |_____^ help: try this: `Ok::<i32, i32>(42).is_err()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching.rs:68:5
   |
LL | /     match Err::<i32, i32>(42) {
LL | |         Ok(_) => false,
LL | |         Err(_) => true,
LL | |     };
   | |_____^ help: try this: `Err::<i32, i32>(42).is_err()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:73:5
   |
LL | /     match Err::<i32, i32>(42) {
LL | |         Ok(_) => true,
LL | |         Err(_) => false,
LL | |     };
   | |_____^ help: try this: `Err::<i32, i32>(42).is_ok()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:78:5
   |
LL | /     match Some(42) {
LL | |         Some(_) => true,
LL | |         None => false,
LL | |     };
   | |_____^ help: try this: `Some(42).is_some()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:83:5
   |
LL | /     match None::<()> {
LL | |         Some(_) => false,
LL | |         None => true,
LL | |     };
   | |_____^ help: try this: `None::<()>.is_none()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:88:13
   |
LL |       let _ = match None::<()> {
   |  _____________^
LL | |         Some(_) => false,
LL | |         None => true,
LL | |     };
   | |_____^ help: try this: `None::<()>.is_none()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:93:20
   |
LL |     let _ = if let Ok(_) = Ok::<usize, ()>(4) { true } else { false };
   |             -------^^^^^--------------------- help: try this: `if Ok::<usize, ()>(4).is_ok()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:96:20
   |
LL |     let x = if let Some(_) = opt { true } else { false };
   |             -------^^^^^^^------ help: try this: `if opt.is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:102:20
   |
LL |     let _ = if let Some(_) = gen_opt() {
   |             -------^^^^^^^------------ help: try this: `if gen_opt().is_some()`

error: redundant pattern matching, consider using `is_none()`
  --> $DIR/redundant_pattern_matching.rs:104:19
   |
LL |     } else if let None = gen_opt() {
   |            -------^^^^------------ help: try this: `if gen_opt().is_none()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching.rs:106:19
   |
LL |     } else if let Ok(_) = gen_res() {
   |            -------^^^^^------------ help: try this: `if gen_res().is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching.rs:108:19
   |
LL |     } else if let Err(_) = gen_res() {
   |            -------^^^^^^------------ help: try this: `if gen_res().is_err()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:141:19
   |
LL |         while let Some(_) = r#try!(result_opt()) {}
   |         ----------^^^^^^^----------------------- help: try this: `while r#try!(result_opt()).is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:142:16
   |
LL |         if let Some(_) = r#try!(result_opt()) {}
   |         -------^^^^^^^----------------------- help: try this: `if r#try!(result_opt()).is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:148:12
   |
LL |     if let Some(_) = m!() {}
   |     -------^^^^^^^------- help: try this: `if m!().is_some()`

error: redundant pattern matching, consider using `is_some()`
  --> $DIR/redundant_pattern_matching.rs:149:15
   |
LL |     while let Some(_) = m!() {}
   |     ----------^^^^^^^------- help: try this: `while m!().is_some()`

error: aborting due to 29 previous errors

